### ISSUE TITLE
errors in other files are reported when running mypy on a file with lowercase drive letter

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#18590]
```py
# foo.py
from sandpit.bar import asdf
```
```py
# bar.py
asdf: int = ""
```
```
> mypy c:\Users\user\project\sandpit\foo.py
c:\Users\user\project\sandpit\bar.py:1: error: Incompatible types in assignment (expression has type "str", variable has type "int")  [assignment]
Found 1 error in 1 file (checked 1 source file)
> mypy C:\Users\user\project\sandpit\foo.py
Success: no issues found in 1 source file
```

**Comment by A5rocks** [comment]
In general mypy fails at case sensitivity on Windows (https://github.com/python/mypy/issues/12880, https://github.com/python/mypy/issues/18445, both have information for if someone wants to solve this) but since this is a specific case I won't close as duplicate.

(If I had to guess, the import is resolved to an upper case drive letter which passes some case-sensitive filter based on input files)

**Comment by DetachHead** [comment]
this causes issues with the mypy vscode plugin which always passes files with a lowercase drive letter. if there are too many errors in other files it won't show any of the errors on the current file

**Comment by wyattscarpenter** [comment]
On mypy 1.15.0, I seem to get `bar.py:1: error: Incompatible types in assignment (expression has type "str", variable has type "int")  [assignment]` no matter which case c has.
